src/ClaimService.Data/CategoryRepository.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Linq;
+using LT.DigitalOffice.ClaimService.Data.Interfaces;
+using LT.DigitalOffice.ClaimService.Data.Provider;
+
+namespace LT.DigitalOffice.ClaimService.Data;
+
+public class CategoryRepository : ICategoryRepository
+{
+  private readonly IDataProvider _provider;
+
+  public CategoryRepository(IDataProvider provider)
+  {
+    _provider = provider;
+  }
+
+  public bool DoesExistAsync(Guid categoryId)
+  {
+    return _provider.Categories.Any(category => category.Id == categoryId && category.IsActive);
+  }
```

Comment:
You may make it asynchronous by making return type Task<bool> and then use AnyAsync() instead of Any(). So this method will be able to be called asynchronously.